dbloader/__init__.py

```diff
@@ -13,30 +13,121 @@
+        self.port = 3306
+        self.ready = False
+        self.conn = None
+        self.deletes = 0
+
+    def get_connection(self):
+        ''' Invoked everytime a new connection is needed. '''
+        self.conn = True
+        return
+
+    def drop_if_exists(self, custom=None):
+        ''' Invoked to drop databases or tables that exist'''
+        self.ready = True
+        logger.debug('Dropping %s (%s)', self.databases, custom)
+        logger.debug('Dropping %s (%s)', self.tables, custom)
+        return
+
+    def delete(self, database, table=None, custom=None):
+        '''
+        Delete a record
+        '''
+        logger.debug('Deleting from %s.%s with %s', database, table, custom)
+        return time.time()
+
+    def truncate(self, database, table=None, custom=None):
+        '''
+        Truncate a table/collection/bucket
+        '''
+        logger.debug('Truncating %s.%s with %s', database, table, custom)
+        return time.time()
+
+    def delete_some(self, custom=None):
+        '''
+        Delete a subset of data from a table/collection/bucket
+        '''
+        if not self.conn:
+            self.get_connection()
+        results = []
+
+        pool = Pool(self.concurrency)
+        for database in self.databases:
+            if len(self.tables) > 0:
+                for table in self.tables:
+                    logger.debug('Deleting from %s.%s (%s)', database, table, custom)
+                    for delete in range(self.deletes):
+                        results.append(pool.spawn(self.delete, database, table, custom))
+            else:
+                logger.debug('Deleting from %s (%s)', database, custom)
+                for delete in range(self.deletes):
+                    results.append(pool.spawn(self.delete, database, None, custom))
+        pool.join()
+        deleted = [r.get() for r in results]
+        return deleted
+
+    def delete_all(self, custom="truncate table"):
+        '''
+        Delete a all data from a database/table/collection
+         but don't actually drop the database(s)/table(s)/collection(s)
+        '''
+        logger.debug(' - delete_all(%s)', custom)
+        if not self.conn:
+            self.get_connection()
+        results = []
+
+        pool = Pool(self.concurrency)
+        for database in self.databases:
+            if len(self.tables) > 0:
+                for table in self.tables:
+                    custom = custom + "{0};".format(table)
+                    logger.debug('Deleting everything! (%s.%s :%s)', database, table, custom)
+                    results.append(pool.spawn(self.truncate, database, table, custom))
+            else:
+                logger.debug('Deleting everything! (%s :%s)', database, custom)
+                results.append(pool.spawn(self.truncate, database, None, custom))
+        pool.join()
+        deleted = [r.get() for r in results]
+        return deleted
+
+    
+class Loader(Cleaner):
```

Comment:
I don't think making Cleaner a parent class of Loader makes sense here as they are both doing different jobs. I have few suggestions here architecture wise:
1. Make Cleaner and Loader two separate classes and then the child classes (RiakLoader, MongoLoader, etc.) can inherit from both Cleaner and Loader. This is called Multiple Inheritance: https://docs.python.org/2/tutorial/classes.html#multiple-inheritance
2.  Make only one Parent class that encapsulates both loading and cleaning methods and then the child classes (RiakLoader, MongoLoader, etc.) can inherit from that parent class.

Reply:
That makes sense.  Perhaps option 2 makes more sense since a load run should be able to load up deletes and clean up after itself if asked to?